.github/actions/analyze/action.yml

```diff
@@ -58,6 +61,46 @@ inputs:
 runs:
   using: composite
   steps:
+
+  - name: Get latest commit hash from master
+    id: latest_commit
+    run: |
+      git fetch origin master
+      LATEST_COMMIT=$(git rev-parse origin/master)
+      echo "LATEST_COMMIT=$LATEST_COMMIT" >> $GITHUB_ENV
+    shell: bash
+
+  - name: Check if previous artifact exists
+    id: artifact_check
+    run: |
+      ARTIFACT_EXISTS=$(curl -s -H "Authorization: Bearer ${{ github.token }}" "https://api.github.com/repos/${{ github.repository }}/actions/artifacts?name=reports-${{ env.LATEST_COMMIT }}" | jq '.total_count' | awk '{print $1}')
+      if [[ $ARTIFACT_EXISTS -gt 0 ]]; then
+        echo "ARTIFACT_EXISTS=true" >> $GITHUB_ENV
+      else
+        echo "ARTIFACT_EXISTS=false" >> $GITHUB_ENV
+      fi
+    shell: bash
+
+  - name: Get artifact URL
+    id: get_artifact_url
+    run: |
+      ARTIFACT_URL=$(curl -s -H "Authorization: Bearer ${{ inputs.GITHUB_TOKEN }}" \
+        "https://api.github.com/repos/${{ github.repository }}/actions/artifacts?name=reports-${{ env.LATEST_COMMIT }}" | \
+        jq -r '.artifacts[0].archive_download_url')
+      echo "ARTIFACT_URL=$ARTIFACT_URL" >> $GITHUB_ENV
+    shell: bash
+
+  - name: Download artifact
+    if: env.ARTIFACT_URL != 'null'
+    run: |
+      curl -L -H "Authorization: Bearer ${{ inputs.GITHUB_TOKEN }}" \
+        "${{ env.ARTIFACT_URL }}" \
```

Comment:
Same for this
```suggestion
        "$ARTIFACT_URL" \
```

Reply:
This LATEST_COMMIT is taken from [this step](https://github.com/binary-com/deriv-app/pull/15172/files#diff-407accfedbc8dedafa97892cca68d7b11e1329aa1f8e836fb54967be01eb73faR90). I think `env.` is needed here.